pydmd/havok.py

```diff
@@ -0,0 +1,194 @@
+        :rtype: numpy.ndarray
+        """
+        return self._embeddings[:,:-1]
+
+    @property
+    def forcing_input(self):
+        """
+        Get the time-delay embedding that forces the linear embeddings.
+
+        :return: array containing the chaotic forcing term.
+        :rtype: numpy.ndarray
+        """
+        return self._embeddings[:,-1]
+
+    @property
+    def A(self):
+        """
+        Get the matrix A in the relationship dV/dt = AV + Bu, where V denotes
+        the linear embeddings and u denotes the forcing input.
+
+        :return: linear dynamics matrix A.
+        :rtype: numpy.ndarray
+        """
+        return self._A
+
+    @property
+    def B(self):
+        """
+        Get the vector B in the relationship dV/dt = AV + Bu, where V denotes
+        the linear embeddings and u denotes the forcing input.
+
+        :return: forcing dynamics vector B.
+        :rtype: numpy.ndarray
+        """
+        return self._B
+
+    def _dehankel(self, X):
+        """
+        Given a hankel matrix X as a 2-D numpy.ndarray,
+        returns the data as a 1-D time-series.
+        """
+        return np.concatenate((X[0, :], X[1:, -1]))
+
+    def reconstructions_of_timeindex(self, timeindex=None):
+        raise NotImplementedError(
+            "This function is not compatible with HAVOK."
+        )
+
+    @property
+    def reconstructed_data(self):
+        # Build a continuous-time system of the following form, where 
+        # x, u, and y represent the states, inputs, and outputs respectively.
+        #
+        # dx/dt = Ax + Bu
+        # y = Cx + Du
+        #
+        # Note that we define the output y to be the state x.
+        havok_system = signal.StateSpace(
+            self.A, self.B, np.eye(self._r-1), 0*self.B
+        )
+
+        # Using the system defined above, compute the output of the system,
+        # where the system input is the forcing input, the times of forcing
+        # are given by self.dmd_timesteps, and the initial state values are
+        # given by the initial values of the linear time-delay embeddings.
+        # This yields a reconstruction of V from the SVD of the hankel matrix.
+        reconstructed_v = signal.lsim(
+            havok_system,
+            U=self.forcing_input,
+            T=self.dmd_timesteps[:len(self.forcing_input)],
+            X0=self.linear_embeddings[0, :]
+        )[1]
+
+        # Compute a reconstruction of the original data x by first recomputing
+        # the hankel matrix with the reconstructed V matrix and then recovering
+        # a 1-D time-series from the computed hankel matrix.
+        reconstructed_hankel_matrix = np.linalg.multi_dot([
+            self._svd_modes[:, :-1],
+            np.diag(self._svd_amps[:-1]),
+            reconstructed_v.conj().T
+        ])
+        reconstructed_x = self._dehankel(reconstructed_hankel_matrix)
+
+        return reconstructed_x
```

Comment:
I propose to merge these just to make it more compact

```suggestion
        return self._dehankel(reconstructed_hankel_matrix)
```